core/src/main/java/tech/tablesaw/io/jdbc/SqlResultSetReader.java

```diff
@@ -89,8 +94,43 @@ public static Table read(ResultSet resultSet, String tableName) throws SQLExcept
         // Setup the columns and add to the table
         for (int i = 1; i <= metaData.getColumnCount(); i++) {
             String name = metaData.getColumnName(i);
-
-            ColumnType type = SQL_TYPE_TO_TABLESAW_TYPE.get(metaData.getColumnType(i));
+            int columnType = metaData.getColumnType(i);
+            ColumnType type = SQL_TYPE_TO_TABLESAW_TYPE.get(columnType);
+            // Try to improve on the initial type assigned to 'type' to minimize size/space of type needed.
+            // For all generic numeric columns inspect closer, checking the precision and 
+            // scale to more accurately determine the appropriate java type to use.
+            if (columnType == Types.NUMERIC || columnType == Types.DECIMAL) {
+                int s = metaData.getScale(i);
+                // When scale is 0 then column is a type of integer
+                if (s == 0) {
+                    int p = metaData.getPrecision(i);
+/* Mapping to java integer types based on integer precision defined:
+
+Java type           TypeMinVal              TypeMaxVal          p               MaxIntVal
+-----------------------------------------------------------------------------------------
+byte, Byte:         -128                    127                 NUMBER(2)       99
+short, Short:       -32768                  32767               NUMBER(4)       9_999
+int, Integer:       -2147483648             2147483647          NUMBER(9)       999_999_999 
+long, Long:         -9223372036854775808    9223372036854775807 NUMBER(18)      999_999_999_999_999_999
+
+*/
+                    // Start with SHORT (since ColumnType.BYTE isn't supported yet)
+                    // and find the smallest java integer type that fits
+                    if (p <= 4) {
+                        type = ColumnType.SHORT;
+                    } else if (p <= 9) {
+                        type = ColumnType.INTEGER;
+                    } else if (p <= 18) {
+                        type = ColumnType.LONG;
+                    }
+                } else { // s is not zero, so a decimal value is expected. First try float, then double
+//                    if (s <= 7) {
```

Comment:
Why is this commented out?

Reply:
I didn't see the same support for FloatColumn that I saw for DoubleColumn in the NumberFilters class.  Specifically when I was trying to call this code:

`			Table filteredTemperatureInCoeffTempRange = model.where(
model.floatColumn("Q_TEMP").isGreaterThanOrEqualTo(model.floatColumn("Q_TEMP_LB"))
.and(model.floatColumn("Q_TEMP").isLessThan(model.floatColumn("Q_TEMP_UB"))));
`
it noted a compile time error that the FloatColumn is not an acceptable input to the isGreaterThanOrEqualTo method.  I see that the type 'double' is an acceptable input to that method, but that doesn't help me with the 'where' method.  My take was that FloatColumn support was not yet present in the NumberFilters class, so commenting out this condition seemed appropriate until FloatColumn was supported by NumberFilters.  

Should I delete the lines instead of commenting them out?  The general goal of this PR is to help interpret the Oracle NUMBER types, and as my changes are they only omit support for the Float type for now.  I was thinking that at some point the commented lines would be able to be reinstated once more FloatColumn support was implemented.

Reply:
@gregorco see my response here: https://github.com/jtablesaw/tablesaw/issues/385#issuecomment-435558189

I've updated `NumberFilters` so that the example you shared here should work. If you use the latest version v0.30.0 then I expect you won't have a problem

Reply:
Awesome.  I'll give it a try.  Thanks.

Reply:
Just submitted the changes.  Made the changes to cleanup ColumnTest.java, and reinstated the lines that were commented out in SqlResultSetReader.java.  As you said, the recent updates to NumberFilter.java resolved the issues I had seen before, and now my code can successfully refer to the FloatColumn types imported from the Oracle query.